tests/test_inspect_pkg.py

```diff
@@ -0,0 +1,26 @@
+# Copyright (C) 2014 Anaconda, Inc
+# SPDX-License-Identifier: BSD-3-Clause
+from __future__ import annotations
+
+import pytest
+from conda import __version__ as conda_version
+from conda.core.prefix_data import PrefixData
+from packaging.version import Version, parse
+
+from conda_build.inspect_pkg import which_package
+
+if TYPE_CHECKING := False:
+    from conda.testing import TmpEnvFixture
+
+
+@pytest.mark.skipif(
+    parse(conda_version) < Version("23.5.0"),
+    reason="tmp_env fixture first available in conda 23.5.0",
+)
+def test_which_package(tmp_env: TmpEnvFixture):
+    with tmp_env("ca-certificates") as prefix:
+        pd = PrefixData(prefix)
+        prec = pd.get("ca-certificates")
+        precs = list(which_package(prefix / prec.files[0], prefix))
```

Comment:
To ensure that the test works in the future, we should either pin this to a certain build of `ca-certificates` or just compose a self-contained test environment here à la
```python
conda_meta = Path(tmp_dir, "conda-meta")
conda_meta.mkdir()
(conda_meta / "history").touch()
(conda_meta / "test-which_package-1-0.json").write_text(prefix_record_json)
```
with `prefix_record_json` being
```json
{
  "build": "0",
  "build_number": 0,
  "channel": "test-which_package-channel",
  "files": [
    "hardlink",
    "softlink"
  ],
  "name": "test-which_package",
  "paths_data": {
    "paths": [
      {
        "_path": "hardlink",
        "path_type": "hardlink",
        "size_in_bytes": 0
      },
      {
        "_path": "softlink",
        "path_type": "softlink",
        "size_in_bytes": 0
      }
    ],
    "paths_version": 1
  },
  "version": "1"
}
```
(or just a copy of a `conda-meta/ca-certificates-*.json`).

Reply:
good idea! reworked this in https://github.com/conda/conda-build/pull/5108/commits/2ce0e20ecf6603d9104d166a4c6ea4e26647449d